design/kata-api-design.md

```diff
@@ -0,0 +1,143 @@
+# Kata API Design
+To fulfill the [kata design requirements](kata-design-requirements.md), and based on the disscusion on [Virtcontainers API extentions](https://docs.google.com/presentation/d/1dbGrD1h9cpuqAPooiEgtiwWDGCYhVPdatq7owsKHDEQ), the kata runtime library features following APIs:
+-  Sandbox based top API
+-  Storage and network hotplug API
+-  Plugin frameworks for external proprietary Kata runtime extensions
+-  Built-in shim and proxy types and capabilities
+
+## Sandbox Based API
+### Sandbox Management API
+-  CreateSandbox(ID, SandboxConfig)
+
+Create, start a sandbox and return the sandbox structure.
+-  ConnectSandbox(ID)
```

Comment:
@amshinde We want it to maintain an agent connection because otherwise most sandbox operation APIs would require the runtime to open a new connection to the agent, do the actual work and then close the connection, which is a waste of CPU/time. It does not need to be really stateful because multiple `FetchPod()` has the same effect and we do not need to actively close the connection. The only requirement is that `FetchPod()` being called before sandbox operations, which is the same in current VC workflow.